### PR TITLE
Correct shared library suffix on macOS

### PR DESCRIPTION
**Pull request by keithc-ca**
Should fix failure seen in https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_extended.system_aarch64_mac_Release/86.

### CODE REVIEW

**Comment by pshipton** [comment]
I'm not sure why we need this or what side affects it might have. There is no failure in the head stream.
https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_extended.system_aarch64_mac_Release/86 is failing on the 0.48 release because https://github.com/eclipse-openj9/openj9/pull/20202 isn't delivered there yet.

File: openj9.test.sharedClasses.jvmti/src/test.sharedClasses.jvmti/net/openj9/stf/SharedClassesAPI.java [hunk 1]
```diff
@@ -213,8 +213,15 @@ public void execute(StfCoreExtension test, StfSharedClassesExtension sharedClass
 				//    due to: https://github.com/eclipse-openj9/openj9-systemtest/issues/38 
 				if ( !PlatformFinder.isWindows() ) {
 					// Verify caches using a JVMTI native agent
-					String nativeExt    =  PlatformFinder.isWindows() ? ".dll" : ".so";
-					String nativePrefix =  PlatformFinder.isWindows() ? "" : "lib";
+					String nativeExt;
+					if (PlatformFinder.isOSX()) {
+						nativeExt = ".dylib";
+					} else if (PlatformFinder.isWindows()) {
```
Comment (pshipton) [inline openj9.test.sharedClasses.jvmti/src/test.sharedClasses.jvmti/net/openj9/stf/SharedClassesAPI.java:219]: The whole block is inside `!PlatformFinder.isWindows()` so checking `PlatformFinder.isWindows() doesn't seem useful.

**Comment by pshipton** [comment]
This change breaks the test on Mac. I will revert it.
`Agent library /Users/jenkins/workspace/Grinder/jvmtest/system/openj9-systemtest/openj9.test.sharedClasses.jvmti/bin/native/osx/libsharedClasses.dylib could not be opened (dlopen(/Users/jenkins/workspace/Grinder/jvmtest/system/openj9-systemtest/openj9.test.sharedClasses.jvmti/bin/native/osx/libsharedClasses.dylib, 1): image not found)`

**Comment by pshipton** [comment]
Reverted via https://github.com/adoptium/openj9-systemtest/pull/159

**Comment by keithc-ca** [comment]
This change was incomplete. I'll create a proper replacement shortly.

**Comment by pshipton** [comment]
Pls make new changes as draft until they are tested.

**Pull request merged**